File: gachisa-backend/src/main/java/com/gachisa/user/dto/UserMeResponse.java
package com.gachisa.user.dto;

import java.time.LocalDateTime;

public record UserMeResponse(
    Long id,
    String email,
    String name,
    String role,
    LocalDateTime createdAt
) {
    public static UserMeResponse from(UserInfo userInfo) {
        return new UserMeResponse(
            userInfo.id(),
            userInfo.email(),
            userInfo.name(),
            userInfo.role().name(),
            userInfo.createdAt()
        );
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/user/dto/UserMeResponse.kt
package com.gachisa.user.dto

import java.time.LocalDateTime

data class UserMeResponse(
    val id: Long,
    val email: String?,
    val name: String,
    val role: String,
    val createdAt: LocalDateTime,
    val status: String,
) {
    companion object {
        @JvmStatic
        fun from(userInfo: UserInfo): UserMeResponse {
            return UserMeResponse(
                id = userInfo.id,
                email = userInfo.email,
                name = userInfo.name,
                role = userInfo.role.name,
                createdAt = userInfo.createdAt,
                status = userInfo.status.name,
            )
        }
    }
}
